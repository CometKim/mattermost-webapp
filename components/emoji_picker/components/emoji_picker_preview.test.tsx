// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import EmojiPickerPreview, {SystemEmoji, CustomEmoji} from './emoji_picker_preview';

describe('components/EmojiPicker/components/EmojiPickerPreview', () => {
    it('should match snapshot with no emoji', () => {
        const wrapper = shallow(<EmojiPickerPreview/>);
        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot with system emoji', () => {
        const emoji: SystemEmoji = {
            aliases: ['grinning'],
            category: 'people',
            batch: '1',
            filename: '1f600',
            offset: null,
            visible: false,
        };
        const wrapper = shallow(<EmojiPickerPreview emoji={emoji}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should match snapshot with custom emoji', () => {
        const emoji: CustomEmoji = {
            aliases: ['test'],
            category: 'custom',
            filename: 'c4d3e4c85frmbfpirk3ctr7opw',
            id: 'c4d3e4c85frmbfpirk3ctr7opw',
            name: 'test',
            offset: null,
            visible: false,
        };
        const wrapper = shallow(<EmojiPickerPreview emoji={emoji}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
