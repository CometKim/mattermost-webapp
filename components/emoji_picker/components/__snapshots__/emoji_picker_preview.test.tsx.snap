// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/EmojiPicker/components/EmojiPickerPreview should match snapshot with custom emoji 1`] = `
<div
  className="emoji-picker__preview"
>
  <div
    className="emoji-picker__preview-image-box"
  >
    <img
      alt="emoji preview image"
      className="emoji-picker__preview-image"
      id="emojiPickerSpritePreview"
      src="/api/v4/emoji/c4d3e4c85frmbfpirk3ctr7opw/image"
    />
  </div>
  <div
    className="emoji-picker__preview-image-label-box"
  >
    <span
      className="emoji-picker__preview-name"
    >
      test
    </span>
    <span
      className="emoji-picker__preview-aliases"
      id="emojiPickerAliasesPreview"
    >
      :test:
    </span>
  </div>
</div>
`;

exports[`components/EmojiPicker/components/EmojiPickerPreview should match snapshot with no emoji 1`] = `
<div
  className="emoji-picker__preview emoji-picker__preview-placeholder"
>
  <FormattedMessage
    defaultMessage="Emoji Picker"
    id="emoji_picker.emojiPicker"
    values={Object {}}
  />
</div>
`;

exports[`components/EmojiPicker/components/EmojiPickerPreview should match snapshot with system emoji 1`] = `
<div
  className="emoji-picker__preview"
>
  <div
    className="emoji-picker__preview-image-box"
  >
    <span
      className="sprite-preview"
    >
      <img
        alt="emoji category image"
        className="emojisprite-preview emoji-category-people-1 emoji-1f600"
        id="emojiPickerSpritePreview"
        src={null}
      />
    </span>
  </div>
  <div
    className="emoji-picker__preview-image-label-box"
  >
    <span
      className="emoji-picker__preview-name"
    >
      grinning
    </span>
    <span
      className="emoji-picker__preview-aliases"
      id="emojiPickerAliasesPreview"
    >
      :grinning:
    </span>
  </div>
</div>
`;
